compiler/damlc/tests/daml-test-files/PreludeTest.daml

```diff
@@ -387,3 +389,18 @@ testDateOverflow = scenario do
 
 testDateUnderflow = scenario do
     pure $ date 2000 Jan 0
+
+testDateRange = scenario do
+    assert (length [(date 2019 Jan 1)..(date 2020 Dec 31)] == 731)
```

Comment:
Let's also test `enumFromThen` with an increasing and a decreasing range and as well as `enumFrom`.
```suggestion
    length [date 2019 Jan 1 .. date 2020 Dec 31] === 731
    [date 1 Jan 6, date 1 Jan 4 ..] === [date 1 Jan 6, date 1 Jan 4, date 1 Jan 2]
    [date 9999 Dec 24, date 9998 Dec 27 ..] === [date 9999 Dec 24, date 9999 Dec 27, date 9999 30]
    [date 9999 Dec 30 ..] === [date 9999 Dec 30, date 9999 Dec 21]
```